valhalla/midgard/openlr.h

```diff
@@ -160,24 +198,36 @@ struct LineLocation {
     attribute1 = raw[index++];
     auto attribute4 = raw[index++];
 
-    last = {longitude, latitude, attribute1, attribute4};
+    lrps.emplace_back(longitude, latitude, attribute1, attribute4);
 
     // Offsets
     std::bitset<8> flags(attribute4);
-    poff = (index < size) && flags[6] ? first.distance * raw[index++] / 256 : 0.f;
-    noff = (index < size) && flags[5] ? first.distance * raw[index++] / 256 : 0.f;
+    poff = (index < size) && flags[6] ? lrps.front().distance * raw[index++] / 256 : 0.f;
+    noff = (index < size) && flags[5] ? std::next(lrps.rbegin())->distance * raw[index++] / 256 : 0.f;
+  }
+
+  LineLocation(const std::vector<LocationReferencePoint>& lrps,
```

Comment:
this is the constructor that allows you to make your own openlr record. since positive and negative offsets are stored in in the LineLocation we need thsi constructor to validate them. ill add a unit test for that as well